<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ParticleEditDialog</class>
 <widget class="QWidget" name="ParticleEditDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>535</width>
    <height>645</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>515</width>
        <height>625</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0" colspan="2">
        <widget class="SimpleTreeWidget" name="layerList">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <column>
          <property name="text">
           <string>Layers</string>
          </property>
         </column>
        </widget>
       </item>
       <item row="0" column="2" rowspan="2">
        <widget class="QGroupBox" name="groupBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>4</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Layer edit</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="9" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Texture</string>
            </property>
           </widget>
          </item>
          <item row="11" column="1" colspan="2">
           <widget class="ColorEditWidget" name="color2" native="true"/>
          </item>
          <item row="10" column="1" colspan="2">
           <widget class="ColorEditWidget" name="color1" native="true"/>
          </item>
          <item row="12" column="1" colspan="2">
           <widget class="ColorEditWidget" name="color3" native="true"/>
          </item>
          <item row="9" column="2">
           <widget class="QPushButton" name="pushButton_3">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="10" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Color 1</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Local velocity</string>
            </property>
           </widget>
          </item>
          <item row="12" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Color 3</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Random velocity</string>
            </property>
           </widget>
          </item>
          <item row="13" column="0">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Color 4</string>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Size grow</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Emission volume</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QSpinBox" name="particleNum">
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QDoubleSpinBox" name="sizeGrow">
            <property name="minimum">
             <double>-100.000000000000000</double>
            </property>
            <property name="maximum">
             <double>100.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Size (minimum, maximum)</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Life (minimum, maximum)</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1">
           <widget class="QLineEdit" name="texture"/>
          </item>
          <item row="11" column="0">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>Color 2</string>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Particle number limit</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>Emission rate</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QDoubleSpinBox" name="emissionRate"/>
          </item>
          <item row="13" column="1" colspan="2">
           <widget class="ColorEditWidget" name="color4" native="true"/>
          </item>
          <item row="6" column="1">
           <widget class="VecEditWidget" name="size" native="true"/>
          </item>
          <item row="5" column="1">
           <widget class="VecEditWidget" name="life" native="true"/>
          </item>
          <item row="4" column="1" colspan="2">
           <widget class="VecEditWidget" name="emissionVol" native="true"/>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="VecEditWidget" name="randomVel" native="true"/>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="VecEditWidget" name="localVel" native="true"/>
          </item>
          <item row="2" column="1" colspan="2">
           <widget class="QCheckBox" name="worldSpaceSim">
            <property name="text">
             <string>Simulate in world space</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButton">
         <property name="text">
          <string>New layer</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_2">
         <property name="text">
          <string>Delete Layer</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ColorEditWidget</class>
   <extends>QWidget</extends>
   <header>coloreditwidget.h</header>
   <container>1</container>
   <slots>
    <signal>valueChanged()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>VecEditWidget</class>
   <extends>QWidget</extends>
   <header>veceditwidget.h</header>
   <container>1</container>
   <slots>
    <signal>valueChanged()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>SimpleTreeWidget</class>
   <extends>QTreeWidget</extends>
   <header>simpletreewidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onNewLayer()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>625</y>
    </hint>
    <hint type="destinationlabel">
     <x>74</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onDeleteLayer()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>186</x>
     <y>625</y>
    </hint>
    <hint type="destinationlabel">
     <x>223</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>localVel</sender>
   <signal>valueChanged()</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>505</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>336</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>randomVel</sender>
   <signal>valueChanged()</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>505</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>emissionVol</sender>
   <signal>valueChanged()</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>505</x>
     <y>234</y>
    </hint>
    <hint type="destinationlabel">
     <x>452</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>life</sender>
   <signal>valueChanged()</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>424</x>
     <y>284</y>
    </hint>
    <hint type="destinationlabel">
     <x>508</x>
     <y>1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>size</sender>
   <signal>valueChanged()</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>424</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>190</x>
     <y>163</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sizeGrow</sender>
   <signal>valueChanged(double)</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>424</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>190</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>particleNum</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>424</x>
     <y>386</y>
    </hint>
    <hint type="destinationlabel">
     <x>193</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>texture</sender>
   <signal>textChanged(QString)</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>424</x>
     <y>413</y>
    </hint>
    <hint type="destinationlabel">
     <x>534</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>color1</sender>
   <signal>valueChanged()</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>505</x>
     <y>465</y>
    </hint>
    <hint type="destinationlabel">
     <x>534</x>
     <y>321</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>color3</sender>
   <signal>valueChanged()</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>505</x>
     <y>565</y>
    </hint>
    <hint type="destinationlabel">
     <x>534</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>color4</sender>
   <signal>valueChanged()</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>505</x>
     <y>615</y>
    </hint>
    <hint type="destinationlabel">
     <x>534</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onPickTexture()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>505</x>
     <y>415</y>
    </hint>
    <hint type="destinationlabel">
     <x>534</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>emissionRate</sender>
   <signal>valueChanged(double)</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>424</x>
     <y>184</y>
    </hint>
    <hint type="destinationlabel">
     <x>534</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>color2</sender>
   <signal>valueChanged()</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>505</x>
     <y>515</y>
    </hint>
    <hint type="destinationlabel">
     <x>534</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>worldSpaceSim</sender>
   <signal>toggled(bool)</signal>
   <receiver>ParticleEditDialog</receiver>
   <slot>onParamChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>389</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>201</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onNewLayer()</slot>
  <slot>onDeleteLayer()</slot>
  <slot>onParamChanged()</slot>
  <slot>onPickTexture()</slot>
 </slots>
</ui>
